api/observability/v1/input_types.go

```diff
@@ -0,0 +1,294 @@
+	// Name used to refer to the input of a `pipeline`.
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Type of output sink.
+	//
+	// +kubebuilder:validation:Required
+	Type InputType `json:"type"`
+
+	// Application, named set of `application` logs that
+	// can specify a set of match criteria
+	//
+	// +kubebuilder:validation:Optional
+	// +nullable
+	Application *Application `json:"application,omitempty"`
+
+	// Infrastructure, Enables `infrastructure` logs.
+	//
+	// +kubebuilder:validation:Optional
+	Infrastructure *Infrastructure `json:"infrastructure,omitempty"`
+
+	// Audit, enables `audit` logs.
+	//
+	// +kubebuilder:validation:Optional
+	Audit *Audit `json:"audit,omitempty"`
+
+	// Receiver to receive logs from non-cluster sources.
+	//
+	// +kubebuilder:validation:Optional
+	Receiver *ReceiverSpec `json:"receiver,omitempty"`
+}
+
+type ContainerInputTuningSpec struct {
+
+	// RateLimitPerContainer is the limit applied to each container
+	// by this input. This limit is applied per collector deployment.
+	//
+	// +kubebuilder:validation:Optional
+	RateLimitPerContainer *LimitSpec `json:"rateLimitPerContainer,omitempty"`
+}
+
+// ApplicationSource defines the type of ApplicationSource log source to use.
+//
+// +kubebuilder:validation:Enum:=container
+type ApplicationSource string
+
+const (
+
+	// ApplicationSourceContainer are container logs from deployed workloads
+	// in any of the following namespaces: default, kube*, openshift*
+	ApplicationSourceContainer ApplicationSource = "container"
+)
+
+// Application log selector.
+// All conditions in the selector must be satisfied (logical AND) to select logs.
+type Application struct {
+	// Selector for logs from pods with matching labels.
+	// Only messages from pods with these labels are collected.
+	// If absent or empty, logs are collected regardless of labels.
+	//
+	// +kubebuilder:validation:Optional
+	Selector *metav1.LabelSelector `json:"selector,omitempty"`
+
+	// Tuning is the container input tuning spec for this container sources
+	//
+	// +kubebuilder:validation:Optional
+	Tuning *ContainerInputTuningSpec `json:"tuning,omitempty"`
+
+	// Includes is the set of namespaces and containers to include when collecting logs.
+	// Note: infrastructure namespaces are still excluded for "*" values unless a qualifying glob pattern is specified.
+	//
+	// +kubebuilder:validation:Optional
+	Includes []NamespaceContainerSpec `json:"includes,omitempty"`
+
+	// Excludes is the set of namespaces and containers to ignore when collecting logs.
+	// Takes precedence over Includes option.
+	//
+	// +kubebuilder:validation:Optional
+	Excludes []NamespaceContainerSpec `json:"excludes,omitempty"`
+}
+
+type NamespaceContainerSpec struct {
+
+	// Namespace specs the namespace from which to collect logs
+	// Supports glob patterns and presumes "*" if omitted.
+	//
+	// +kubebuilder:validation:Optional
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
Generally good to add length limits and specific character sets to strings where possible